proposals/lip-0057.md

```diff
@@ -1349,8 +1350,12 @@ def getActiveValidators(validatorsTwoRoundsAgo: list[EligibleValidatorObject], r
 
         # Determine active validators and their BFT weight.
         activeValidators = [{"address": item.address, "bftWeight": ceiling(item.weight, WEIGHT_SCALE_FACTOR) } for item in validatorsTwoRoundsAgo[:nbrElectedValidators]]
-        averageElectedValidatorBftWeight = sum([item.bftWeight for item in activeValidators]) // nbrElectedValidators
-        activeValidators+= [{"address": item, "bftWeight": averageElectedValidatorBftWeight} for item in chosenInitValidators]
+        if len(activeValidators) > 0:
+            averageElectedValidatorBftWeight = sum([item.bftWeight for item in activeValidators]) // len(activeValidators)
```

Comment:
Question: could this value still be 0 if `activeValidators` contains elements with `bftWeight = 0`?

Reply:
Good question. In general it could be possible, but for our PoS, to become eligible you need at least weight 1000, therefore the bftweight would be at least 1. So no elements of this array should have `bftWeight = 0`.